libqfieldsync/project_checker.py

```diff
@@ -439,3 +444,70 @@ def check_external_layers(
             )
 
         return None
+
+    def check_layer_geopackage_actions(
+        self, layer: LayerSource
+    ) -> Optional[FeedbackResult]:
+        """Check if layers from the same GeoPackage have offline and copy actions."""
+        if not layer.is_file or not layer.filename:
+            return None
+
+        # Check if is GeoPackage layer
+        if not layer.filename.lower().endswith(".gpkg"):
+            return None
+
+        current_action = layer.action
+        if current_action not in [SyncAction.OFFLINE, SyncAction.COPY]:
+            return None
+
+        # Group all layers by their GeoPackage file
+        gpkg_layers = {}
+        for project_layer in self.project.mapLayers().values():
+            source = LayerSource(project_layer)
+            if not source.is_file or not source.filename:
+                continue
+
+            if not source.filename.lower().endswith(".gpkg"):
+                continue
+
+            if source.action not in [SyncAction.OFFLINE, SyncAction.COPY]:
+                continue
+
+            if source.filename not in gpkg_layers:
+                gpkg_layers[source.filename] = []
+            gpkg_layers[source.filename].append((project_layer.name(), source.action))
+
+        # Check for GeoPackages with mixed actions
+        if layer.filename in gpkg_layers and len(gpkg_layers[layer.filename]) > 1:
+            has_offline = any(
+                action == SyncAction.OFFLINE
+                for _, action in gpkg_layers[layer.filename]
+            )
+            has_copy = any(
+                action == SyncAction.COPY for _, action in gpkg_layers[layer.filename]
+            )
```

Comment:
```
for filename, layer_sources in layer_sources_by_filename.items():
  has_mixed_actions = False
  first_layer_action = layer_sources[0].action

  for layer_source in layer_sources:
    if layer_source.action !=  first_layer_action:
      has_mixed_actions = True
      break
```

Reply:
This will make the code slightly easier to follow instead of `any` with inline for loops in it.